src/libraries/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTraceSymbols.cs

```diff
@@ -63,7 +63,7 @@ void IDisposable.Dispose()
             {
                 Handle handle = MetadataTokens.Handle(methodToken);
 
-                if (handle.Kind == HandleKind.MethodDefinition)
+                if (!handle.IsNil && handle.Kind == HandleKind.MethodDefinition)
```

Comment:
We can do this check even before `TryGetReader`. No point in getting the reader if we are not going to use it anyway.